src/Sarif/Schemata/sarif-schema.json

```diff
@@ -1388,6 +1373,53 @@
       "required": ["message"]
     },
 
+    "resultProvenance": {
+      "description": "Contains information about how and when a result was detected.",
+      "additionalProperties": false,
+      "type": "object",
+      "properties": {
+
+        "firstDetectionTimeUtc": {
+          "description": "The Coordinated Universal Time (UTC) date and time at which the result was first detected. See \"Date/time properties\" in the SARIF spec for the required format.",
+          "type": "string",
+          "format": "date-time"
+        },
+
+        "lastDetectionTimeUtc": {
+          "description": "The Coordinated Universal Time (UTC) date and time at which the result was most recently detected. See \"Date/time properties\" in the SARIF spec for the required format.",
+          "type": "string",
+          "format": "date-time"
+        },
+
+        "firstDetectionRunInstanceGuid": {
+          "description": "A GUID-valued string equal to the id.instanceGuid property of the run in which the result was first detected.",
+          "type": "string"
+        },
+
+        "lastDetectionRunInstanceGuid": {
+          "description": "A GUID-valued string equal to the id.instanceGuid property of the run in which the result was most recently detected.",
+          "type": "string"
+        },
+
+        "invocationIndex": {
+          "description": "The index within the run.invocations array of the invocation object which describes the tool invocation that detected the result.",
+          "type": "integer",
+          "minimum": -1
+        },
+
+        "conversionSources": {
+          "description": "An array of physicalLocation objects which specify the portions of an analysis tool's output that a converter transformed into the result.",
+          "type": "array",
+          "minItems": 0,
+          "uniqueItems": true,
+          "default": [],
```

Comment:
>"default": [] [](start = 10, length = 13)

why are you providing this default? we don't do this for other arrays. is is providing some benefit? if so, we should scrub the schema for consistency.

Reply:
I searched before doing this. There is exactly one other place we do this for an empty array (propertyBag.tags). I didn't know what we wanted to settle on (especially since our array scrub explicitly stated that for many of the 42 array-valued properties, the default really is "[]". I'll go whichever way you want on this.

---
In reply to: [244648474](https://github.com/Microsoft/sarif-sdk/pull/1193#discussion_r244648474) [](ancestors = 244648474)

Reply:
Removed.

---
In reply to: [244648526](https://github.com/Microsoft/sarif-sdk/pull/1193#discussion_r244648526) [](ancestors = 244648526,244648474)